### ISSUE TITLE
LOWESS regression should allow arbitrary x-values

### ISSUE THREAD

**Issue by tgbrooks** [statsmodels/statsmodels#6543]
#### Is your feature request related to a problem? Please describe
Currently `statsmodels.nonparametric.smoothers_lowess.lowess` takes `endog` and `exog` arrays and returns fitted values at the same `exog` values as provided, or alternatively, at a subset of those values if `delta` is provided. However lowess regression is well-defined at every potential `x`-value and it should be easy for a user to perform lowess regression at a given set of `x`-values, just as say a linear model provides a `predict()` function for computing regressed values at arbitrary points.

One compelling use-case is providing boot-strap confidence bounds to a lowess regression. One might ressample their `exog` and `endog` data and rerun the lowess. However, then the output at each ressample will give the regressed values at a different set of `x`-values and cannot easily be compared to find confidence bounds.

#### Describe the solution you'd like
The simplest change to support this would be to allow a `x` parameter to `statsmodels.nonparametric.smoothers_lowess.lowess` which determines where the regression is evaluated. If `x` is passed in, then `delta` is ignored and a one-dimensional `out` array is returned with the regression values at the corresponding values of `x`. If `return_sorted = True` then the `x` would be sorted, the function would return a two-dimensional `out` array with the sorted `x` values and their corresponding regressed values.

#### Describe alternatives you have considered
The proposed solution above, however, is inconsistent with the `predict()`-style interface of `OLS` and other models. However, the lowess fit interface is already inconsistent with those so I see no uniform way to fit both styles together, so I propose the simpler interface.

#### Additional context
A [Stack Overflow question](https://stackoverflow.com/questions/36252434/predicting-on-new-data-using-locally-weighted-regression-loess-lowess) addresses this problem but has no completely satisfactory answer. The top answer performs a lowess fit and then interpolates that to get values at the given `x`-coordinates which will be practically useful but unsatisfactory and more work when compared to the proposed change.

Would a pull request implementing this feature likely be accepted?

**Comment by bashtage** [comment]
A well thought out request and a useful feature.

**Comment by tgbrooks** [comment]
I drafted a PR for this issue. It should provide the interface described above. I've never submitted a PR before, so I left it as draft for now. Comments are welcome.

Some notes:
-  I named the new optional parameter `xvals`. Any other suggestions? I'm not sold on it.
- `return_sorted` is ignored if `xvals` is provided.
- `is_sorted = True` requires the new `xvals` to also be sorted if provided
- NaN values in `xvals` are given NaNs in their regressed values
- I added three tests
- There is likely a minor performance regression in the non-`xvals` case, though I didn't measure it. Weights are now computed and used always where before they were not used or computed when unnecessary.
- Occasionally some data points may fail regression due to all weights being 0 of nearby points, The existing function fills those points with their original value, i..e does not perform regression. However, this is not possible when specifying `xvals` since there is no original y-value. Instead, I chose to fill those with NaNs. Arguably, it would be better to fill them with NaNs even when `xvals` is not provided, but I did not want to change the bevahior for any existing calls.
